src/main/java/gyro/google/cloudsql/DbPasswordValidationPolicy.java

```diff
@@ -0,0 +1,151 @@
+
+    private String complexity;
+    private Boolean disallowUsernameSubstring;
+    private Boolean enablePasswordPolicy;
+    private Integer minLength;
+    private String passwordChangeInterval;
+    private Integer reuseInterval;
+
+    /**
+     * The complexity of the password.
+     */
+    @ValidStrings({ "COMPLEXITY_DEFAULT", "COMPLEXITY_UNSPECIFIED" })
+    @Updatable
+    public String getComplexity() {
+        return complexity;
+    }
+
+    public void setComplexity(String complexity) {
+        this.complexity = complexity;
+    }
+
+    /**
+     * When set to ``true``, disallows username as a part of the password.
+     */
+    @Updatable
+    public Boolean getDisallowUsernameSubstring() {
+        return disallowUsernameSubstring;
+    }
+
+    public void setDisallowUsernameSubstring(Boolean disallowUsernameSubstring) {
+        this.disallowUsernameSubstring = disallowUsernameSubstring;
+    }
+
+    /**
+     * When set to ``true``, password policy is enabled.
+     */
+    @Updatable
+    @Required
+    public Boolean getEnablePasswordPolicy() {
+        return enablePasswordPolicy;
+    }
+
+    public void setEnablePasswordPolicy(Boolean enablePasswordPolicy) {
+        this.enablePasswordPolicy = enablePasswordPolicy;
+    }
+
+    /**
+     * The minimum number of characters allowed.
+     */
+    @Updatable
+    public Integer getMinLength() {
+        return minLength;
+    }
+
+    public void setMinLength(Integer minLength) {
+        this.minLength = minLength;
+    }
+
+    /**
+     * The minimum interval after which the password can be changed.
+     * Only for PostgresSQL.
+     */
+    @Updatable
+    public String getPasswordChangeInterval() {
+        return passwordChangeInterval;
+    }
+
+    public void setPasswordChangeInterval(String passwordChangeInterval) {
+        this.passwordChangeInterval = passwordChangeInterval;
+    }
+
+    /**
+     * The number of previous passwords that cannot be reused.
+     */
+    @Updatable
+    public Integer getReuseInterval() {
+        return reuseInterval;
+    }
+
+    public void setReuseInterval(Integer reuseInterval) {
+        this.reuseInterval = reuseInterval;
+    }
+
+    @Override
+    public void copyFrom(PasswordValidationPolicy model) throws Exception {
+        setComplexity(model.getComplexity());
+        setDisallowUsernameSubstring(model.getDisallowUsernameSubstring());
+        setEnablePasswordPolicy(model.getEnablePasswordPolicy());
+        setMinLength(model.getMinLength());
+        setPasswordChangeInterval(model.getPasswordChangeInterval());
+        setReuseInterval(model.getReuseInterval());
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    public PasswordValidationPolicy toPasswordValidationPolicy() {
+        PasswordValidationPolicy policy = new PasswordValidationPolicy();
+        policy.setEnablePasswordPolicy(getEnablePasswordPolicy());
```

Comment:
Is the rest of the configuration still allowed if this is set to false ?

Reply:
Added validation